core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -452,6 +452,7 @@ object TestUtils extends Logging {
     } catch {
       case e: ExecutionException => if (!(e.getCause != null &&
```

Comment:
Why do we want to throw exception when we have a same number of partitions and replicas, and do not throw exception otherwise?

Reply:
Yes, I have the same confusion at first. But look again, it'll find there is a (not)! in front of the whole condition. That is:
**!** (e.getCause != null &&
          e.getCause.isInstanceOf[TopicExistsException] &&
          !waitForAllPartitionsMetadata(brokers, topic, effectiveNumPartitions).isEmpty &&
          topicHasSameNumPartitionsAndReplicationFactor(admin, topic,
            effectiveNumPartitions, replicationFactor)
)

Reply:
Can we change the condition to be clearer?

Reply:
Yes, PR updated to make the condition clearer